inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+plotting and `lm()` function for linear modeling.
+4. Add the regression line to the scatter plot for visualization.
+5. Examine the slope, intercept, and R-squared value to 
+understand the relationship between the variables.
+6. Test the linear modeling assumptions.
+7. Treat other possible covariates.
+8. Perform visualizations and interpret the analysis outcomes. You can use
+visualization methods from packages like miaViz, scater, etc.
+
+```{r}
+#| label: extra-regressionchart
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# Load the necessary libraries
+library(ggplot2)
+library(TreeSummarizedExperiment)
+library(mia)
+library(miaViz)
+library(scater)
+
+# Load the demo dataset 
+data("GlobalPatterns", package = "mia")
+tse <- GlobalPatterns
+
+# Calculate the Shannon diversity index
+diversity <- estimateDiversity(tse, assay.type = "counts", index = "shannon")
+
+# Extract the Shannon diversity index and bmi_group
+shannon_index <- diversity$shannon
+bmi_group <- colData(tse)$SampleType  # Using SampleType as a proxy for BMI_Group
+
+# Create the data frame
+data <- data.frame(
+  Shannon_Index = shannon_index,
+  BMI_Group = bmi_group
+)
+
+# Perform the regression analysis
+model <- lm(Shannon_Index ~ BMI_Group, data = data)
+
+# Create the plot with the regression line
+p <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index)) +
+  geom_point(color = 'blue') +
+  geom_smooth(method = 'lm', color = 'red') +
+  labs(title = 'Regression Analysis of Shannon Index vs. BMI Group',
+       x = 'BMI Group', y = 'Shannon Index') +
+  theme_minimal()
+
+# Print the plot
+print(p)
+
+# Print the model summary
+summary(model)
+```
+What do we interpret from the above regression analysis?
+
+1. The coefficients of the regression model indicate the relationship 
+between the BMI groups and the Shannon diversity index. If the coefficient
+for a particular BMI group is significantly different from zero, it
+suggests that this group has a different Shannon diversity index compared
+to the reference group.
+2. The p-value associated with each coefficient tells us whether the effect
+of that BMI group on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+3. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the BMI groups. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
+
+```{r}
+#| label: sub-checking-modeling-assumptions
+
+# 1. Linearity
+plot(model, which = 1)
+
+# 2. Normality of residuals
+plot(model, which = 2)
+
+# 3. Homoscedasticity
+plot(model, which = 3)
+
+# 4. Independence of residuals (Durbin-Watson test)
+dwtest(model)
+```
+Here, 
+1. Linearity plot(Plot 1) shows the residuals vs. fitted values. The residuals
+should be randomly scattered around zero, indicating that the linearity
```

Comment:
Either explain each of these as separate code examples, or organize all subplots on a single row.